docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+// fewer Gateways than IPs.
+Gateways []net.IP
+
+// GatewayIPv6LLA is the IPv6 Link Local Address for the pod's gateway, that is the address
+// that will be set as gateway with router advertisements
+// generated from the gateway router from the node where the pod is running.
+GatewayIPv6LLA net.IP
+
+// Routes are additional routes to add to the pod's network namespace
+Routes []PodRoute
+
+// TunnelID assigned to each pod for layer2 secondary networks
+TunnelID int
+
+// Role defines what role this network plays for the given pod.
+// Expected values are:
+// (1) "primary" if this network is the primary network of the pod.
+//     The "default" network is the primary network of any pod usually
+//     unless user-defined-network-segmentation feature has been activated.
+//     If network segmentation feature is enabled then any user defined
+//     network can be the primary network of the pod.
+// (2) "secondary" if this network is the secondary network of the pod.
+//     Only user defined networks can be secondary networks for a pod.
+// (3) "infrastructure-locked" is applicable only to "default" network if
+//     a user defined network is the "primary" network for this pod. This
+//     signifies the "default" network is only used for probing and
+//     is otherwise locked for all intents and purposes.
+// At a given time a pod can have only 1 network with role:"primary"
+Role string
+}
+```
+This annotation will be used to build an initial cache of allocated addresses at startup, which will then be updated
+dynamically at runtime and used for conflict detection.
+A similar approach is required for IP address conflict detection.
+When a conflict is detected the pod should not start and an appropriate event should be emited.
+
+When the `NetworkSelectionElement` contains an `IPAMClaimReference` the referenced IPAMClaim should
+reflect the IP allocation status including error reporting through the newly introduced
+`Conditions` status field.
+In the opposite scenario where the `NetworkSelectionElement` does not specify the `IPAMClaimReference`
+the IP allocation is not persisted when the pod is removed.
+
+### Testing Details
+
+
+The following scenarios should be covered in testing:
+
+* VM workloads import into OVN-Kubernetes with no changes to the instances network configuration.
+* Imported VM workloads can live-migrate to another node without any additional traffic disruption.
+* 'v1.multus-cni.io/default-network' cannot be changed after the pod was created.
+* It should be possible to configure the pods MAC or the IP address without configuring the other.
+* When `excludeSubnets` is configured automatic IP allocation should not use addresses specified in it.
+* It should be possible to configure the pods IP address using the 'v1.multus-cni.io/default-network'
+even if the address is a part of the `excludeSubnets`.
+* Requesting an IP address that is not a part of the networks subnet should fail.
```

Comment:
```suggestion
* Requesting an IP address and default gateway IP that is not a part of the networks subnet should fail.
```

Reply:
hmmm isn't that what I am suggestion to also add? meaning add a test also for ensuring we fail when the "default gatewayIP" that is not part of subnet??

Reply:
+1 to test (thus effectively *document*) we do not support a gateway IP outside the subnet.